### PR TITLE
2LSS and conversions

### PR DESCRIPTION
**Pull request by vischia**
Hi,

this includes Xuan's work on 2LSS, and my preliminary on conversions (I need the new samples to finish it).

This also includes PRs #57 and #58.

Best,
Pietro

### CODE REVIEW

**Comment by vischia** [comment]
Including the latest commits from @cheidegg , and some additional work by @cericeci for the 2lss

File: TTHAnalysis/macros/leptons/friendChunkCheck.sh
Comment (cheidegg) [inline TTHAnalysis/macros/leptons/friendChunkCheck.sh:18]: Why is removing the underscore necessary? Can't you just give "evVarFriend" instead of "evVarFriend_" as an argument?

File: TTHAnalysis/macros/verifyAllFTrees.sh
Comment (cheidegg) [inline TTHAnalysis/macros/verifyAllFTrees.sh]: Not sure what this script is good for at all since it anway calls the verifyFTree with the same arguments as passed to this script. Do we really need it in the repository? If you just wanna save the default values, you could keep it in a local script outside of the repo.

File: TTHAnalysis/python/plotter/functions.cc
Comment (cheidegg) [inline TTHAnalysis/python/plotter/functions.cc]: AFAIK all this stuff is deprecated and should already have been moved to separate files like functionsEWK.cc functions2LSS.cc. Also the TTH stuff. Please check with @peruzzim @folguera  
Especially triggerSF and leptonSF are in susy-ewkino/functionsSF.cc and PU weights are in susy-ewkino/functionsPUW.cc

File: TTHAnalysis/python/plotter/mcPlots.py
Comment (cheidegg) [inline TTHAnalysis/python/plotter/mcPlots.py:1192]: Is this necessary or can't you just change the directory name on command line?

File: TTHAnalysis/python/plotter/susy-ewkino/2lss/systs_ewkino_2lss.txt
Comment (cheidegg) [inline TTHAnalysis/python/plotter/susy-ewkino/2lss/systs_ewkino_2lss.txt]: Why a separate systs file? The idea was to use the common one in susy-ewkino/systs_ewkino.txt

File: TTHAnalysis/python/plotter/susy-ewkino/3l/functionsEWK.cc
Comment (cheidegg) [inline TTHAnalysis/python/plotter/susy-ewkino/3l/functionsEWK.cc]: Does this CR call need to be in the SR call? 
This is super dangerous to edit anything in this function (actually in the entire file) and I'd prefer to leave this function untouched and put it in a separate macro in the dedicated CR directory.

File: TTHAnalysis/python/tools/leptonBuilderEWK.py
Comment (cheidegg) [inline TTHAnalysis/python/tools/leptonBuilderEWK.py:654]: Can you please leave these commented lines for reference (as soon as we are close to final reading, at least)?

**Review by cheidegg — CHANGES_REQUESTED** [review]
Several fixes required. The functions.cc has old functions that already were moved into different macros some time ago. Please do not re-introduce them. Also the directory susy-ewkino/3l/taus  could go into susy-ewkino, like "susy-ewkino/taustudy" ?

File: TTHAnalysis/macros/leptons/friendChunkCheck.sh
Comment (vischia) [reply TTHAnalysis/macros/leptons/friendChunkCheck.sh:18]: Sure. Done.

File: TTHAnalysis/macros/verifyAllFTrees.sh
Comment (vischia) [reply TTHAnalysis/macros/verifyAllFTrees.sh]: It's not for the default values. I can even remove them.

But the verifyFTree as it is requires a single dataset name, while this script lets you specify a whole folder, which is far quicker and often what you actually want to do. 

I think it is an useful addition.

File: TTHAnalysis/python/plotter/functions.cc
Comment (vischia) [reply TTHAnalysis/python/plotter/functions.cc]: We had agreed that these things would have been removed by @folguera after merging #60, since #60 touches a series of 2LSS stuff.

Leaving untouched.

File: TTHAnalysis/python/plotter/mcPlots.py
Comment (vischia) [reply TTHAnalysis/python/plotter/mcPlots.py:1192]: Short answer: this is unfortunately still necessary.
Long answer: for now we are stuck with this system in which we have to read using /pool/ciencias/ and write using /pool/cienciasrw/ (the latter being the same partition mounted R/W but on a far slower connection than the one mounted in /pool/ciencias). We will test with a lustre system probably in January. I know it is crazy, but this is what we have now.

Comment (vischia) [reply TTHAnalysis/python/plotter/mcPlots.py:1192]: (Leaving unchanged)

File: TTHAnalysis/python/plotter/susy-ewkino/2lss/systs_ewkino_2lss.txt
Comment (vischia) [reply TTHAnalysis/python/plotter/susy-ewkino/2lss/systs_ewkino_2lss.txt]: @GonzalezFJR @cericeci can answer.

If they don't answer shortly, I suggest to leave that for after @folguera 's #60

File: TTHAnalysis/python/plotter/susy-ewkino/3l/functionsEWK.cc
Comment (vischia) [reply TTHAnalysis/python/plotter/susy-ewkino/3l/functionsEWK.cc]: Done.

**Comment by vischia** [comment]
Hi,

I have answered everything (some rejected some accepted) and pushed the relevant commits. I had clicked something about a review, and canceled because I was not sure it was supposed to be clicked :dancer:  


P.S. Now that I found the smilies here, you are all doomed.

**Review by vischia — COMMENTED** [review]
Ah this is it: I suppose like that I am requesting the review of my answers to the review... :interrobang:

File: TTHAnalysis/python/plotter/functions.cc
Comment (vischia) [reply TTHAnalysis/python/plotter/functions.cc]: Speaking with @nachosandres actually the situation seems to be that this stuff had been removed from functions.cc but not put anywhere else, so its inclusion back has been an emergency patch to have stuff ready for the FSR!

(still suggesting to leave it untouched until #60 is merged, as agreed on skype)

File: TTHAnalysis/macros/verifyAllFTrees.sh
Comment (cheidegg) [reply TTHAnalysis/macros/verifyAllFTrees.sh]: python verifyFTree.py (treeDir) (friendDir)
will loop over all samples in (treeDir)

File: TTHAnalysis/python/plotter/mcPlots.py
Comment (cheidegg) [reply TTHAnalysis/python/plotter/mcPlots.py:1192]: OK we can leave it for now. 
When doing a PR to central, I expect that we'll be asked to remove it since it is something very specific in a very general script.

File: TTHAnalysis/macros/verifyAllFTrees.sh
Comment (vischia) [reply TTHAnalysis/macros/verifyAllFTrees.sh]: Uuuh Nice! I did not notice! I will take the script out then!

File: TTHAnalysis/python/plotter/susy-ewkino/2lss/cuts_2lss.txt
Comment (folguera) [inline TTHAnalysis/python/plotter/susy-ewkino/2lss/cuts_2lss.txt:16]: this should be left here.... they are useful

File: TTHAnalysis/python/plotter/susy-ewkino/2lss/cuts_2lss_dphimu.txt
Comment (folguera) [inline TTHAnalysis/python/plotter/susy-ewkino/2lss/cuts_2lss_dphimu.txt]: this file should be removed

File: TTHAnalysis/python/plotter/susy-ewkino/2lss/cuts_baseline.txt
Comment (folguera) [inline TTHAnalysis/python/plotter/susy-ewkino/2lss/cuts_baseline.txt]: this should be renamed to cuts_2lss and added the stuff that I don't want to remove from cuts_2lss

File: TTHAnalysis/python/plotter/susy-ewkino/2lss/cuts_crtnl.txt
Comment (folguera) [inline TTHAnalysis/python/plotter/susy-ewkino/2lss/cuts_crtnl.txt]: remove this file

File: TTHAnalysis/python/plotter/susy-ewkino/2lss/mca-blinded.txt
Comment (folguera) [inline TTHAnalysis/python/plotter/susy-ewkino/2lss/mca-blinded.txt]: removed this mca file

File: TTHAnalysis/python/plotter/susy-ewkino/2lss/mca_2lss_dphimu.txt
Comment (folguera) [inline TTHAnalysis/python/plotter/susy-ewkino/2lss/mca_2lss_dphimu.txt]: remove this as well

File: TTHAnalysis/python/plotter/susy-ewkino/2lss/plots_2lss_dphimu.txt
Comment (folguera) [inline TTHAnalysis/python/plotter/susy-ewkino/2lss/plots_2lss_dphimu.txt]: remove this plots

File: TTHAnalysis/python/plotter/susy-ewkino/2lss/plots_baseline.txt
Comment (folguera) [inline TTHAnalysis/python/plotter/susy-ewkino/2lss/plots_baseline.txt]: remove this plots

File: TTHAnalysis/python/plotter/susy-ewkino/2lss/plots_crtnl.txt
Comment (folguera) [inline TTHAnalysis/python/plotter/susy-ewkino/2lss/plots_crtnl.txt]: this should be renamed to plots_2lss

File: TTHAnalysis/python/plotter/susy-ewkino/2lss/systs_ewkino_2lss.txt
Comment (folguera) [inline TTHAnalysis/python/plotter/susy-ewkino/2lss/systs_ewkino_2lss.txt]: remove this file

File: TTHAnalysis/macros/verifyAllFTrees.sh
Comment (vischia) [reply TTHAnalysis/macros/verifyAllFTrees.sh]: Done.

File: TTHAnalysis/python/plotter/functions.cc
Comment (vischia) [reply TTHAnalysis/python/plotter/functions.cc]: Blanked the file taking it from the current ECOP 80X_M17 head, as agreed on skype.

File: TTHAnalysis/python/plotter/mcPlots.py
Comment (vischia) [reply TTHAnalysis/python/plotter/mcPlots.py:1192]: I switched to setting an --env variable at runtime. This final solution should be fine also for central, as it is the same that has been adopted in susy-interface, no?  (if it is not, then central should review its way of dealing with people having other computing centers than cern :octopus:  )

**Comment by vischia** [comment]
Hi,

I cleaned up the tau directories, and renamed the currently used one to susy-ewkino/crtau.

File: TTHAnalysis/python/plotter/susy-ewkino/2lss/cuts_2lss.txt
Comment (vischia) [reply TTHAnalysis/python/plotter/susy-ewkino/2lss/cuts_2lss.txt:16]: So no action needed?

File: TTHAnalysis/python/plotter/susy-ewkino/2lss/cuts_2lss_dphimu.txt
Comment (vischia) [reply TTHAnalysis/python/plotter/susy-ewkino/2lss/cuts_2lss_dphimu.txt]: Done

File: TTHAnalysis/python/plotter/susy-ewkino/2lss/cuts_crtnl.txt
Comment (vischia) [reply TTHAnalysis/python/plotter/susy-ewkino/2lss/cuts_crtnl.txt]: Done.

File: TTHAnalysis/python/plotter/susy-ewkino/2lss/cuts_2lss.txt
Comment (vischia) [reply TTHAnalysis/python/plotter/susy-ewkino/2lss/cuts_2lss.txt:16]: I saw now the comment below. OK so I merge the two files, basically, giving priority to the content of the baseline one, correct?

File: TTHAnalysis/python/plotter/susy-ewkino/2lss/mca-blinded.txt
Comment (vischia) [reply TTHAnalysis/python/plotter/susy-ewkino/2lss/mca-blinded.txt]: Done

File: TTHAnalysis/python/plotter/susy-ewkino/2lss/mca_2lss_dphimu.txt
Comment (vischia) [reply TTHAnalysis/python/plotter/susy-ewkino/2lss/mca_2lss_dphimu.txt]: LoL. Done.

File: TTHAnalysis/python/plotter/susy-ewkino/2lss/plots_2lss_dphimu.txt
Comment (vischia) [reply TTHAnalysis/python/plotter/susy-ewkino/2lss/plots_2lss_dphimu.txt]: Do you mean only L1, or the full file containing 3 plots?

File: TTHAnalysis/python/plotter/susy-ewkino/2lss/plots_baseline.txt
Comment (vischia) [reply TTHAnalysis/python/plotter/susy-ewkino/2lss/plots_baseline.txt]: The full file, no?

File: TTHAnalysis/python/plotter/susy-ewkino/2lss/plots_crtnl.txt
Comment (vischia) [reply TTHAnalysis/python/plotter/susy-ewkino/2lss/plots_crtnl.txt]: There is already a plots_2lss file. Could you please confirm that I should just rename susy-ewkino/2lss/plots_crtnl.txt overwriting the current plots_2lss.txt?

File: TTHAnalysis/python/plotter/susy-ewkino/2lss/systs_ewkino_2lss.txt
Comment (vischia) [reply TTHAnalysis/python/plotter/susy-ewkino/2lss/systs_ewkino_2lss.txt]: Done.

File: TTHAnalysis/python/plotter/susy-ewkino/2lss/plots_2lss_dphimu.txt
Comment (folguera) [reply TTHAnalysis/python/plotter/susy-ewkino/2lss/plots_2lss_dphimu.txt]: the full file out

File: TTHAnalysis/python/plotter/susy-ewkino/2lss/plots_baseline.txt
Comment (folguera) [reply TTHAnalysis/python/plotter/susy-ewkino/2lss/plots_baseline.txt]: yes

File: TTHAnalysis/python/plotter/susy-ewkino/2lss/plots_crtnl.txt
Comment (folguera) [reply TTHAnalysis/python/plotter/susy-ewkino/2lss/plots_crtnl.txt]: I confirm

File: TTHAnalysis/python/plotter/susy-ewkino/2lss/cuts_2lss.txt
Comment (folguera) [reply TTHAnalysis/python/plotter/susy-ewkino/2lss/cuts_2lss.txt:16]: I mean that all the lines that are commented and removed in this file should be available in the new merged file.  sorry if I was not clear

Comment (vischia) [reply TTHAnalysis/python/plotter/susy-ewkino/2lss/cuts_2lss.txt:16]: OK! Sorry for asking, it is just to make sure :) 

Concerning the exclusive cut, is it OK to leave the _Recl uncommented and the _Mini commented out, or should I restore the _Mini as the one active?

\# exclusive: nLepTight_Mini<3 && nTightTauSel==0
exclusive: nLepTight_Recl<3 && nTightTauSel==0

File: TTHAnalysis/python/plotter/susy-ewkino/2lss/plots_2lss_dphimu.txt
Comment (vischia) [reply TTHAnalysis/python/plotter/susy-ewkino/2lss/plots_2lss_dphimu.txt]: Done.

File: TTHAnalysis/python/plotter/susy-ewkino/2lss/plots_baseline.txt
Comment (vischia) [reply TTHAnalysis/python/plotter/susy-ewkino/2lss/plots_baseline.txt]: Done

File: TTHAnalysis/python/plotter/susy-ewkino/2lss/plots_crtnl.txt
Comment (vischia) [reply TTHAnalysis/python/plotter/susy-ewkino/2lss/plots_crtnl.txt]: Done

File: TTHAnalysis/python/plotter/susy-ewkino/2lss/cuts_2lss.txt
Comment (vischia) [reply TTHAnalysis/python/plotter/susy-ewkino/2lss/cuts_2lss.txt:16]: @nachosandres tells me that in 2lSS it is used normally _Recl, so I am leaving _Recl for the time being. In case it it not the correct choice, just let me know before approving the PR :)

File: TTHAnalysis/python/plotter/susy-ewkino/2lss/cuts_baseline.txt
Comment (vischia) [reply TTHAnalysis/python/plotter/susy-ewkino/2lss/cuts_baseline.txt]: Done

File: TTHAnalysis/python/plotter/susy-ewkino/2lss/cuts_2lss.txt
Comment (vischia) [reply TTHAnalysis/python/plotter/susy-ewkino/2lss/cuts_2lss.txt:16]: Done

**Comment by vischia** [comment]
Dear @folguera @cheidegg , everything that we discussed this morning is now part of this PR :)

**Review by cheidegg — APPROVED** [review]

**Comment by cheidegg** [comment]
I leave it to @folguera to check the 2lss part.

**Comment by folguera** [comment]
should be fine... merging

**Pull request merged**
